newIDE/app/src/ObjectEditor/Editors/SpriteEditor/AnimationPreview.js

```diff
@@ -164,17 +190,46 @@ const AnimationPreview = ({
 
   return (
     <Column expand noOverflowParent noMargin>
-      <ImagePreview
-        resourceName={resourceName}
-        imageResourceSource={getImageResourceSource(resourceName)}
-        isImageResourceSmooth={isImageResourceSmooth(resourceName)}
-        initialZoom={initialZoom}
-        project={project}
-        hideCheckeredBackground={hideCheckeredBackground}
-        hideControls={hideControls}
-        fixedHeight={fixedHeight}
-        fixedWidth={fixedWidth}
-      />
+      <div style={{ position: 'relative' }}>
+        <ImagePreview
+          resourceName={resourceName}
+          imageResourceSource={getImageResourceSource(resourceName)}
+          isImageResourceSmooth={isImageResourceSmooth(resourceName)}
+          initialZoom={initialZoom}
+          project={project}
+          hideCheckeredBackground={hideCheckeredBackground}
+          hideControls={hideControls}
+          fixedHeight={fixedHeight}
+          fixedWidth={fixedWidth}
+          onImageLoaded={() => {
+            imagesLoadedArray.current[currentFrameIndexRef.current] = true;
+            // When the array of loaders changes, decide if we display the loader or not.
+            // If all images are loaded, then hide loader for instant display.
+            const hasFinishedLoadingAllResources = !imagesLoadedArray.current.some(
+              hasImageLoaded => !hasImageLoaded
+            );
+            if (hasFinishedLoadingAllResources) {
+              setIsStillLoadingResources(false);
+            }
+            // Image has loaded, so cancel the timeout if it was set.
+            if (loaderTimeout.current) clearTimeout(loaderTimeout.current);
```

Comment:
Should `loaderTimeout.current` be set back to `null`?

Reply:
entirely right